From b31a87f9477456f4f9c212a47096aa15c56a5f21 Mon Sep 17 00:00:00 2001
From: Ethan <parrotgeek1@gmail.com>
Date: Sun, 29 Jan 2017 16:18:28 -0800
Subject: [PATCH 2/9] Update misc.c

---
 core/misc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/misc.c b/core/misc.c
index 0be513f..23524ff 100644
--- a/core/misc.c
+++ b/core/misc.c
@@ -502,10 +502,10 @@ uint32_t misc_read(uint32_t addr) {
          * match the 80E0 field in an OS image. 01 = CAS, 00 = non-CAS. */
             
         case 0x28: 
-            debugger(DBG_USER, 0);
+            warn("it reads the 900A0028 at PC");
             return idreg[asic_user_flags].lo;
         case 0x2C: 
-            debugger(DBG_USER, 0);
+            warn("it reads the 900A002C at PC");
             return idreg[asic_user_flags].hi;
     }
     return bad_read_word(addr);
-- 
2.11.0 (Apple Git-81)

